### PR TITLE
Sip008 initial support

### LINKED ISSUE #1703: 支持SIP008在线配置
如同 ShadowsocksR 一样，这样当服务器提供方变更服务器地址时可以自动变更本地服务器设定。

### LINKED ISSUE #2928: 增加订阅服务器功能
**Is your feature request related to a problem? Please describe.**
shadowsocks安卓版添加了订阅功能，希望订阅服务器功能，这样可通过局域网同步电脑上的服务器。

### LINKED ISSUE #2939: 4.1.9.3版本的windows如何添加订阅地址
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by studentmain**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

- [x] [Searched](https://github.com/shadowsocks/shadowsocks-windows/search?q=is%3Apr&type=Issues) for similar pull requests
- [x] Compiled the code with Visual Studio
- [x] Require translation update
- [x] Require document update (readme.md, wikipage, etc)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New feature

---

### Description of your *pull request* and other information

Close #1703
Close #2939
Close #2928

Finally, here we are......

Some glitch is expected...

### CODE REVIEW

**Comment by studentmain** [comment]
see https://github.com/shadowsocks/shadowsocks-org/issues/89

**Comment by chenshaoju** [comment]
The quick test is passed,base example fie at https://github.com/shadowsocks/shadowsocks-android/blob/master/.github/doc-json.md#appendix-sample-exported-json-file 。

image


:+1: Looks good to me, but someone else must approve.

**Review by chenshaoju — APPROVED** [review]

File: shadowsocks-csharp/Model/Server.cs [hunk 1]
```diff
@@ -43,6 +43,7 @@ public class Server
         [DefaultValue("")]
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore, DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
         public string plugin_args;
+        // TODO The legacy of the historical version (Required field in SIP003)
```
Comment (studentmain) [inline shadowsocks-csharp/Model/Server.cs:46]: Is `remarks` a required field?

Comment (studentmain) [reply shadowsocks-csharp/Model/Server.cs:46]: according to SIP002 spec: `SS-URI = "ss://" userinfo "@" hostname ":" port [ "/" ] [ "?" plugin ] [ "#" tag ]`

Comment (Stzx) [reply shadowsocks-csharp/Model/Server.cs:46]: From the description of `Basic file format`. Yes

Comment (Stzx) [reply shadowsocks-csharp/Model/Server.cs:46]: In fact, I want to initiate a PR to your fork, but I didn't expect that the Desktop program was pushed directly to the branch.

Comment (studentmain) [reply shadowsocks-csharp/Model/Server.cs:46]: https://github.com/shadowsocks/shadowsocks-android/blob/master/.github/doc-json.md#profile-json-object

> A JSON object is recognized as a `Profile` (i.e. a valid server config), if all of the following required fields are present and valid:
>
> - `server`
> - `server_port`
> - `password`
> - `method`

Comment (Stzx) [reply shadowsocks-csharp/Model/Server.cs:46]: Oh, it looks like the two descriptions conflict.

https://github.com/shadowsocks/shadowsocks-org/issues/89

> ## Basic file format
> Basically, we follows the original JSON config format of shadowsocks. The required fields are:
> 
> ```json
> {
>   "server": "198.199.101.152",
>   "server_port": 8388,
>   "password": "u1rRWTssNv0p",
>   "method": "aes-256-cfb",
>   "remarks": "Example 1"
> }
> ```

Comment (studentmain) [reply shadowsocks-csharp/Model/Server.cs:46]: It's confused, we should discuss it in that issue.

Comment (Stzx) [reply shadowsocks-csharp/Model/Server.cs:46]: Let [doc-json.md](https://github.com/shadowsocks/shadowsocks-android/blob/master/.github/doc-json.md#profile-json-object) prevail. For maximum compatibility

Comment (Stzx) [reply shadowsocks-csharp/Model/Server.cs:46]: You can delete `TODO` and update the basic format , then force push.

Comment (studentmain) [reply shadowsocks-csharp/Model/Server.cs:46]: sounds good.

> In fact, I want to initiate a PR to your fork, but I didn't expect that the Desktop program was pushed directly to the branch.

BTW, I personally recommend GitExtension.

**Review by Stzx — APPROVED** [review]
Other refactoring may be carried out in `v5`.

**Review by database64128 — APPROVED** [review]
I think it's ready for merging.

**Pull request merged**